javascript/src/test/rationals/Rational.test.js

```diff
@@ -97,28 +120,47 @@ describe("Rational class tests", () => {
   describe("times/multiply tests", () => {
     test("(2 pts) if either rational is zero, it gives zero as result", () => {
       const expected = zero;
-      const firstActual = zero.times(one);
-      const secondActual = one.times(zero);
+      const zeroTimesOne = zero.times(one);
+      const oneTimesZero = one.times(zero);
 
-      expect(firstActual).toEqual(expected);
-      expect(secondActual).toEqual(expected);
+      expect(zeroTimesOne).toEqual(expected);
+      expect(oneTimesZero).toEqual(expected);
+
+      // must return a new instance
+      expect(Object.is(zeroTimesOne, zero)).toBe(false);
+      expect(Object.is(oneTimesZero, zero)).toBe(false);
+
+      // must not mutate inputs
+      expect(one).toMatchObject({ numerator: 1, denominator: 1});
+      expect(zero).toMatchObject({ numerator: 0, denominator: 1});
     });
 
     test("(2 pts) 1/3 * 2/3 = 2/9", () => {
       const expected = new Rational(2, 9);
       const actual = oneThird.times(twoThirds);
+
       expect(actual).toEqual(expected);
+
+      // must not mutate inputs
+      expect(one).toMatchObject({ numerator: 1, denominator: 1 });
+      expect(negativeOne).toMatchObject({ numerator: -1, denominator: 1 });
```

Comment:
shouldn't the inputs being checked here be `oneThird` and `twoThirds`?

Reply:
Good catch, fixed